src/Elastic.Apm/AgentComponents.cs

```diff
@@ -47,9 +52,14 @@ public class AgentComponents : IApmAgent, IDisposable
 
 		public void Dispose()
 		{
-			if (PayloadSender is IDisposable disposable)
+			if (PayloadSender is IDisposable disposablePayloadSender)
```

Comment:
I think there is nothing wrong with this. This was already here, it’s the same situation with MetricsCollector.  Reason is that the `IPayloadSender` and `IMetricsCollector` interfaces aren’t `Disposable`, which makes sense, because e.g. the implementations we use in tests don’t need to dispose anything. On the other hand the real implementations need to dispose stuff and the `AgentComponent` knows this. That’s why it’s this way. Happy to follow up, but I’d prefer doing it outside of this PR.

Reply:
I've created issue #260